### PR TITLE
Jakarta RESTful Web Services 3.0 - Spec and Javadocs

### PR DESCRIPTION
**Pull request by spericas**
- [X] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [X] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [X] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [X] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [ ] For a Release Review, a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification.
The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
- [X] The URL of the OSSRH staging repository for the api, javadoc:
https://jakarta.oss.sonatype.org/content/groups/staging/jakarta/ws/rs/jakarta.ws.rs-api/
- [X] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
 https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/jakarta-restful-ws-tck-3.0.0.zip
- [X] The URL of the compatibility certification request issue:
https://github.com/eclipse-ee4j/jaxrs-api/issues/912
- [X] Specification JavaDoc in the wombat/x.y/apidocs directory.

### CODE REVIEW

**Comment by kwsutter** [comment]
@spericas Thanks for creating the draft PR for jax-rs.  I've assigned @starksm64 as the Spec Committee mentor.  He should be reviewing this PR and be your point of contact for getting this ready for ballot.  Thanks.

**Comment by starksm64** [comment]
# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form jakarta-{spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form jakarta-{spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to project release plan of the form https://projects.eclipse.org/projects/ee4j.{spec}/releases/x.y/plan
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip (The folder path is required, the file name pattern is preferred.)
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request
 
10. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Review by starksm64 — CHANGES_REQUESTED** [review]

**Comment by starksm64** [comment]
There is a problem with the staged api javadocs:
https://jakarta.oss.sonatype.org/content/groups/staging/jakarta/ws/rs/jakarta.ws.rs-api/3.0.0/jakarta.ws.rs-api-3.0.0-javadoc.jar

When I download and look at the index.html, even though the title says '... jakarta.ws.rs-api 3.0.0 API', the first h1 element says 'JAX-RS 2.2 API Specification'

File: restful-ws/3.0/_index.md [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
 summary: "Release for Jakarta EE 9"
 ---
 Jakarta RESTful Web Services provides a foundational API to develop web services
 following the Representational State Transfer (REST) architectural pattern.
 
 * [Jakarta RESTful Web Services 3.0 Release Record](https://projects.eclipse.org/projects/ee4j.jaxrs/releases/3.0)
   * [Jakarta EE Platform 9 Release Plan](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan)
-* [Jakarta RESTful Web Services 3.0 Specification Document](./restful-ws-spec-3.0-M1.pdf) (PDF)
-* [Jakarta RESTful Web Services 3.0 Specification Document](./restful-ws-spec-3.0-M1.html) (HTML)
+* [Jakarta RESTful Web Services 3.0 Specification Document](./restful-ws-spec-3.0.pdf) (PDF)
```
[the remark this answers is not in the document]
Comment (spericas) [reply restful-ws/3.0/_index.md:11]: @starksm64 I'll fix the file names. As for the logo, that's a problem, we had issues in the M1 PDF with mathematical symbols and had to generate the PDF from HTML this time around. We'd need to somehow add it to the HTML, which is probably not the intention.

**Comment by spericas** [comment]
> There is a problem with the staged api javadocs:
> https://jakarta.oss.sonatype.org/content/groups/staging/jakarta/ws/rs/jakarta.ws.rs-api/3.0.0/jakarta.ws.rs-api-3.0.0-javadoc.jar
> 
> When I download and look at the index.html, even though the title says '... jakarta.ws.rs-api 3.0.0 API', the first h1 element says 'JAX-RS 2.2 API Specification'

That's odd, I think this issue was resolved (it's OK in my local copy). Maybe the "re-staging" process we have is faulty and we haven't pushed the latest bits to staging. Will take a look.

File: restful-ws/3.0/_index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (spericas) [reply restful-ws/3.0/_index.md:11]: @starksm64 Renamed and manual update of PDF file. Hope it's acceptable now.

**Comment by spericas** [comment]
> > There is a problem with the staged api javadocs:
> > https://jakarta.oss.sonatype.org/content/groups/staging/jakarta/ws/rs/jakarta.ws.rs-api/3.0.0/jakarta.ws.rs-api-3.0.0-javadoc.jar
> > When I download and look at the index.html, even though the title says '... jakarta.ws.rs-api 3.0.0 API', the first h1 element says 'JAX-RS 2.2 API Specification'
> 
> That's odd, I think this issue was resolved (it's OK in my local copy). Maybe the "re-staging" process we have is faulty and we haven't pushed the latest bits to staging. Will take a look.

 The issue with the Staging repo should be fixed now. Link above should point to latest bits now.

**Comment by starksm64** [comment]
Specs and javadoc now look good, so just the TCK and compatible request are needed.

File: restful-ws/3.0/_index.md [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+* [Jakarta RESTful Web Services 3.0 Specification Document](./restful-ws-spec-3.0.html) (HTML)
 * [Jakarta RESTful Web Services 3.0 Javadoc](./apidocs)
-* [Jakarta RESTful Web Services 3.0 TCK]()([sig](),[sha](),[pub]())
+* [Jakarta RESTful Web Services 3.0 TCK](https://download.eclipse.org/jakartaee/restful-ws/3.0/jakarta-restful-ws-tck-3.0.0.zip)([sig](https://download.eclipse.org/jakartaee/restful-ws/3.0/jakarta-restful-ws-tck-3.0.0.zip.sig,[sha](https://download.eclipse.org/jakartaee/restful-ws/3.0/jakarta-restful-ws-tck-3.0.0.zip.sha256),[pub](https://raw.githubusercontent.com/jakartaee/specification-committee/master/jakartaee-spec-committee.pub))
```
[the remark this answers is not in the document]
Comment (spericas) [reply restful-ws/3.0/_index.md]: Fixed.

**Comment by starksm64** [comment]
The TCK looks good. The remaining checklist items require a compatible implementation and certification request. I have added a new set of checklist items for the EMO/PMC notification tasks. Those can be done in parallel.

**Comment by spericas** [comment]
@starksm64 URL to compatibility certification request updated at the top. Release review initiated.

**Comment by starksm64** [comment]
Verified sum matches CCR:
Scotts-iMacPro:tmp starksm$ shasum -a256 jakarta-restful-ws-tck-3.0.0.zip 
39081f393f495d2363ae66c487f83a171cb52a8038930bd1af7829f5757a9aa7  jakarta-restful-ws-tck-3.0.0.zip

**Review by starksm64 — APPROVED** [review]
This looks ready to go to ballot now. I did some updates to the _index.md page to allow the specification preview to look correct.

**Comment by starksm64** [comment]
Start of the voting thread
https://www.eclipse.org/lists/jakarta.ee-spec/msg01136.html

**Review by kwsutter — APPROVED** [review]

**Comment by arjantijms** [comment]
Release bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=567847

**Comment by arjantijms** [comment]
@starksm64 It's been over 14 days now for the ballot. Should it be closed?

**Comment by kwsutter** [comment]
Thanks for the nudge, @arjantijms.  Here's the finalization checklist that @starksm64 will be following to complete this Spec PR.  (I know he's been traveling lately...)

* On ballot completion, the specification committee mentor:
 - [ ] adds this final checklist to the main PR.
 - [ ] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
 \<include url of Issue here>


 ----
 - [ ] promotes api staging release promotes the specification api jars to maven central. An example release job script can be found here https://wiki.eclipse.org/MavenReleaseScript.
 - [ ] go through the merged jakarta.ee specification website page to verify all the links are valid.
 - [ ] approve the compatibility request.
 - [ ] The compatible implementation project/vendor MUST send an email to tck@eclipse.org for approval of the compatible implementation for trademark usage.
 - [ ] merge any final release branch as appropriate for the branch management for the project.

**Comment by arjantijms** [comment]
Ballot completed: https://eclipse.org/lists/jakarta.ee-spec/msg01266.html

**Comment by starksm64** [comment]
Representative | Representative for: | Vote
-- | -- | --
Kenji Kazumura | Fujitsu | +1
Dan Bandera, Kevin Sutter | IBM | +1
Ed Bratt, Dmitry Kornilov | Oracle | +1
Andrew Pielage, Matt Gill | Payara | +1
Scott Stark, Mark Little | Red Hat | +1
David Blevins, Jean-Louis Monteiro, Cesar Hernandez | Tomitribe | +1
Ivar Grimstad | EE4J PMC | +1
Marcelo Ancelmo, Martijn Verburg | Participant Members |  
Werner Keil | Committer Members | +1
Scott (Congquan) Wang | Enterprise Members | +1
  | Total | 9

**Pull request merged**